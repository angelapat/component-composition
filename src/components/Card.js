import React from 'react';
import PropTypes from 'prop-types';

export default function Card({ _id, name, image }) {
  return (
    <div >
      <header><h3>{name}</h3></header>
      <figure>
        <img src={image} alt={name} />
      </figure>
      <p>{name} has {_id}</p>
    </div>
  )
};

Card.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};