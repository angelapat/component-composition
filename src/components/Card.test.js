import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';


describe('Card component', () => {

  it('renders Card', () => {
    const card = shallow(<Card _id={'5da237699734fdcb7bef8f51'} name={'Arnold Shortman'} image={'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'}/>);
    expect(card).toMatchSnapshot();
  });
});