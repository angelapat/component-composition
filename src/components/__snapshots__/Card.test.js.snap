// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card 1`] = `
<div>
  <header>
    <h3>
      Arnold Shortman
    </h3>
  </header>
  <figure>
    <img
      alt="Arnold Shortman"
      src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
    />
  </figure>
</div>
`;
